forgerock-authenticator/src/main/java/org/forgerock/android/authenticator/Push.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.authenticator;
+
+import org.forgerock.android.authenticator.util.SortedList;
+
+import java.util.List;
+
+/**
+ * Represents an instance of a Push authentication mechanism. Associated with an Account.
+ */
+public class Push extends Mechanism {
+
+    /** The endpoint used for registration */
+    public static final String REG_ENDPOINT_KEY = "r";
+    /** The endpoint used for authentication */
+    public static final String AUTH_ENDPOINT_KEY = "a";
+    /** The message id to use for response */
+    public static final String MESSAGE_ID_KEY = "m";
+    /** The shared secret used for signing */
+    public static final String BASE_64_SHARED_SECRET_KEY = "s";
+    /** The challenge to use for the response */
+    public static final String BASE_64_CHALLENGE_KEY = "c";
+    /** The challenge to use for the response */
+    public static final String AM_LOAD_BALANCER_COOKIE_KEY = "l";
+
+
+    /** The registration URL for Push mechanism */
+    private String registrationEndpoint;
+    /** The authentication URL for Push mechanism */
+    private String authenticationEndpoint;
+    /** Notifications objects associated with this push mechanism */
+    private final List<Notification> notifications;
```

Comment:
This is not used, I like the idea of creating relationship between data model, however, it seem the StorageClient interface did not utilize it. The interface defines store and retrieves independently, they will eventually have data integrity issue, for example, we remove all account but not mechanism, or we remove Push Mechanism but did not remove Notification?

Reply:
This is just a placeholder, I was thinking about a Lazy initialization for those objects. I believe iOS implemented that way, but it is not decided yet. We will review it and consider this feedback.